pysat/_instrument.py

```diff
@@ -3287,48 +3380,40 @@ def load(self, yr=None, doy=None, end_yr=None, end_doy=None, date=None,
-            if (not self._empty(self._prev_data)) & (not self.empty):
-                stored_data = self.data  # .copy()
-                temp_time = copy.deepcopy(self.index[0])
-
-                # Pad data using access mechanisms that works for both pandas
-                # and xarray
-                self.data = self._prev_data.copy()
-
-                # __getitem__ used below to get data from instrument object.
-                # Details for handling pandas and xarray are different and
-                # handled by __getitem__.
-                self.data = self[first_pad:temp_time]
-                if not self.empty:
-                    if self.index[-1] == temp_time:
-                        self.data = self[:-1]
-                    self.concat_data(stored_data, prepend=False)
-                else:
-                    self.data = stored_data
-
-            if (not self._empty(self._next_data)) & (not self.empty):
-                stored_data = self.data  # .copy()
-                temp_time = copy.deepcopy(self.index[-1])
-
-                # Pad data using access mechanisms that work for both pandas
-                # and xarray
-                self.data = self._next_data.copy()
-                self.data = self[temp_time:last_pad]
-                if len(self.index) > 0:
-                    if (self.index[0] == temp_time):
-                        self.data = self[1:]
-                    self.concat_data(stored_data, prepend=True)
-                else:
-                    self.data = stored_data
+            if not self._empty(self._prev_data) and not self.empty:
+                # __getitem__ used to handle any pandas/xarray differences in
+                # data slicing
+                pdata = self.__getitem__(slice(first_pad, self.index[0]),
+                                         data=self._prev_data)
+                if not self._empty(pdata):
+                    # Test the data index, slicing if necessary
+                    pindex = self._index(data=pdata)
+                    if pindex[-1] == self.index[0]:
+                        pdata = self.__getitem__(slice(-1), data=pdata)
+                    self.concat_data(pdata, prepend=True)
+                del pdata
+
+            if not self._empty(self._next_data) and not self.empty:
+                # __getitem__ used to handle any pandas/xarray differences in
+                # data slicing
+                ndata = self.__getitem__(slice(self.index[-1], last_pad),
+                                         data=self._next_data)
+                if not self._empty(ndata):
+                    # Test the data index, slicing if necessary
+                    nindex = self._index(data=ndata)
+                    if nindex[0] == self.index[-1]:
+                        ndata = self.__getitem__(
+                            slice(1, len(ndata[self.index.name])), data=ndata)
+                    self.concat_data(ndata, prepend=False)
+                del ndata
```

Comment:
@JonathonMSmith here's the bit of code that I was talking about.  See how much simpler it is now!

Reply:
The old code mentions in the comments that __getitem__ is used but I don't see it used explicitly, was it being used at all before?
Also the boolean for the concat operations have changed. Is that because of all the copying that was going on before. It's much more intuitive that the previous data would be prepended... instead of appending the current data to the previous data lol

Reply:
Yes, `__getitem__` is a magic function, which means it's present in all classes and gets data from the class.  However, it's not meant to be called by a user as a method or function.  So when you do something like `inst['varname']` to get a particular variable from an Instrument, the Instrument class is actually calling `__getitem__`.  When you have a complicated class like `Instrument`, you can write your own `__getitem__` so long as you follow the rules for what this method requires.  In the old code `self.data[:-1]` was actually calling `__getitem__`.

You're right that all the copying going on meant that we needed to switch the way concat combined the data.  Now it's more straightforward.